src/main.cpp

```diff
@@ -55,7 +55,7 @@ int main([[maybe_unused]] int argc, [[maybe_unused]] const char **argv)
   ImGui::GetStyle().ScaleAllSizes(scale_factor);
   ImGui::GetIO().FontGlobalScale = scale_factor;
 
-  std::array<bool, 12> states{};
+  std::array<bool, 13> states{};
```

Comment:
You are correct that `states` needs 13 rather than 12 values.

However, if you used experimental::make_array to make an array of all the strings, you could remove the need to count the number of steps at all.

```
    #include <experimental/array>
    constexpr auto steps = std::experimental::make_array("The Plan",  "Getting Started", "Step3...");
    std::array<bool, steps.size()> states{};
```

Reply:
Well if you really want to go down that path, you could just write `constexpr std::array steps{"The Plan", "Getting Started", "Step3..."};`, you don't need `make_array` for that